app/controllers/customers_controller.rb

```diff
@@ -20,25 +20,32 @@ def new
   # GET /customers/1/edit
   def edit
   end
-
   # POST /customers
   # POST /customers.json
-  def create
-    @customer = Customer.new(customer_params)
 
+  def create
+    if phone_exist(params[:customer][:phone_number])
+      @customer = Customer.find_by(phone_number: params[:customer][:phone_number])
+      flag = true
+    else
+      @customer = Customer.new(customer_params)
```

Comment:
@kalgibhatt-btc If we don't need to put some condition on `.save`, we can do this one one line using `.create`